felt_python/elements.py

```diff
@@ -1,87 +1,162 @@
 """Elements and element groups"""
 
 import json
+from typing import Dict, Any, List, Union
 
 from urllib.parse import urljoin
 
 from .api import make_request, BASE_URL
+from .util import deprecated
 
 
-MAP_ELEMENTS_TEMPLATE = urljoin(BASE_URL, "maps/{map_id}/elements/")
-ELEMENT_TEMPLATE = urljoin(MAP_ELEMENTS_TEMPLATE, "{element_id}")
-MAP_ELEMENT_GROUPS_TEMPLATE = urljoin(BASE_URL, "maps/{map_id}/element_groups/")
-ELEMENT_GROUP_TEMPLATE = urljoin(MAP_ELEMENT_GROUPS_TEMPLATE, "{element_group_id}")
+ELEMENTS = urljoin(BASE_URL, "maps/{map_id}/elements")
+ELEMENT = urljoin(BASE_URL, "maps/{map_id}/elements/{element_id}")
+ELEMENT_GROUPS = urljoin(BASE_URL, "maps/{map_id}/element_groups")
+ELEMENT_GROUP = urljoin(BASE_URL, "maps/{map_id}/element_groups/{element_group_id}")
 
 
 def list_elements(map_id: str, api_token: str | None = None):
-    """List all elements on a map"""
+    """List all elements on a map
+
+    Args:
+        map_id: The ID of the map to list elements from
+        api_token: Optional API token
+
+    Returns:
+        GeoJSON FeatureCollection of all elements
+    """
     response = make_request(
-        url=MAP_ELEMENTS_TEMPLATE.format(map_id=map_id),
+        url=ELEMENTS.format(map_id=map_id),
         method="GET",
         api_token=api_token,
     )
     return json.load(response)
 
 
 def list_element_groups(map_id: str, api_token: str | None = None):
-    """List all element groups on a map"""
+    """List all element groups on a map
+
+    Args:
+        map_id: The ID of the map to list element groups from
+        api_token: Optional API token
+
+    Returns:
+        List of element groups
+    """
     response = make_request(
-        url=MAP_ELEMENT_GROUPS_TEMPLATE.format(map_id=map_id),
+        url=ELEMENT_GROUPS.format(map_id=map_id),
         method="GET",
         api_token=api_token,
     )
     return json.load(response)
 
 
-def list_elements_in_group(
+def show_element_group(
```

Comment:
Is "show" the right term here? This still returns a list, it doesn't print anything. The [developer docs](https://developers.felt.com/rest-api/api-reference/elements#list-all-elements-in-a-group) also say "list" so we should keep these consistent

Reply:
Yeah, `show` isn't correct here. To match all the other endpoint functions, this would be `get_element_group`.

The old `list_elements_in_group` makes sense when you read it, but doesn't follow the naming pattern. For example, we have `get_layer_group` not `list_layers_in_group`.

Also it technically returns a GeoJSON object, not a literal list.

I'll change it to `get_element_group` for now and we can consider if we should keep that or just stick with the old function.